### PR TITLE
Added custom footer text area

### PR DESCRIPTION
**Pull request by kalle-1989**

### CODE REVIEW

File: php/seatreg_functions.php [hunk 1]
```diff
@@ -593,6 +593,21 @@ function seatreg_generate_settings_form() {
 				<textarea class="form-control" id="registration-info-text" name="registration-info-text" placeholder="<?php esc_html_e('Enter info text here', 'seatreg'); ?>"><?php echo esc_html($options[0]->info); ?></textarea>
 			</div>
 
+			<div class="form-group">
+				<label for="custom-footer-text"><?php esc_html_e('Custom footer text', 'seatreg'); ?></label>
```
Comment (SiimKirjanen) [inline php/seatreg_functions.php:597]: Seems like we don't need `for="custom-footer-text"` attribute on the label element. At least when I click on the label it wont autofocus editor

Comment (SiimKirjanen) [inline php/seatreg_functions.php:597]: What do you think about changing the **Custom footer text** label to be something like **Registration custom footer area**?

File: registration/js/registration.js [hunk 2]
```diff
@@ -851,6 +856,12 @@ SeatReg.prototype.generateField = function(fieldName, isLastCheckItem) {
 	return label;
 };
 
+SeatReg.prototype.generateCustomFooterFextField = function (customFooterText) {
+	var decodedcustomFooterText = $("<div/>").html(customFooterText).text();
+	var label = $('<label class="field-label custom-footer-text" data-label="custom-footer-text"><span class="l-text">' + decodedcustomFooterText + '</span></label>');
```
[the remark this answers is not in the document]
Comment (SiimKirjanen) [reply registration/js/registration.js:861]: And we can remove `field-label` class

[the remark this answers is not in the document]
Comment (SiimKirjanen) [reply registration/js/registration.js:861]: I don't think we need `<span class="l-text"></span> `

**Comment by SiimKirjanen** [comment]
Also there is a security concern. For example if you save `<script>alert('Hey');</script>` to the editor. Then the script will execute when the booking modal is open

File: registration/js/registration.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalle-1989) [reply registration/js/registration.js]: Changed: https://github.com/SiimKirjanen/seatreg/commit/b22ca78626a601b99a5e080cdeefd1d1caa9ecb0. I am not an java expert. Please let me know if there is a better way.

File: php/seatreg_functions.php (hunk 1 above)
Comment (kalle-1989) [reply php/seatreg_functions.php]: Agree. Changed: https://github.com/SiimKirjanen/seatreg/commit/565febfd0c182fd810bd494c738c934e40fe049b

File: registration/js/registration.js [hunk 3]
```diff
@@ -851,6 +856,12 @@ SeatReg.prototype.generateField = function(fieldName, isLastCheckItem) {
 	return label;
 };
 
+SeatReg.prototype.generateCustomFooterFextField = function (customFooterText) {
+	var decodedcustomFooterText = $("<div/>").html(customFooterText).text();
+	var div = $('<div class="field-label custom-footer-text">'+ decodedcustomFooterText + '</div>');
+	return div;
```
Comment (SiimKirjanen) [inline registration/js/registration.js:862]: Lets remove script tags before returning the `div`

`div.find('script').remove();`

Comment (kalle-1989) [reply registration/js/registration.js]: Changed: https://github.com/SiimKirjanen/seatreg/commit/f4e645f5ff3754d07f0991e102f06bd28c8651d9

**Review by SiimKirjanen — APPROVED** [review]
Looks good. Thank you for contributing to SeatReg

**Pull request merged**